openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -362,6 +421,61 @@ def pre_requisite_courses(self):
         return json.loads(self._pre_requisite_courses_json)
 
     @classmethod
+    def get_select_courses(cls, course_keys):
+        """
+        Returns CourseOverview objects for the given course_keys.
+        """
+        course_overviews = []
+
+        log.info('Generating course overview for %d courses.', len(course_keys))
+        log.debug('Generating course overview(s) for the following courses: %s', course_keys)
+
+        for course_key in course_keys:
+            try:
+                course_overviews.append(CourseOverview.get_from_id(course_key))
+            except Exception as ex:  # pylint: disable=broad-except
+                log.exception(
+                    'An error occurred while generating course overview for %s: %s',
+                    unicode(course_key),
+                    ex.message,
+                )
+
+        log.info('Finished generating course overviews.')
+
+        return course_overviews
+
+    @classmethod
+    def get_all_courses(cls, force_reseeding=False):
+        """
+        Returns all CourseOverview objects in the database.
+
+        Arguments:
+            force_reseeding (bool): Optional parameter.
+                If True, the modulestore is used as the source of truth for
+                the list of courses, even if the CourseOverview table was
+                previously seeded.
+
+                If False, the list of courses is retrieved from the
+                CourseOverview table if it was previously seeded, falling
+                back to the modulestore if it wasn't seeded.
+        """
+        if force_reseeding or not CourseOverviewGeneratedHistory.objects.first():
+            # Seed the CourseOverview table with data for all
+            # courses in the system.
+            course_keys = [course.id for course in modulestore().get_courses()]
+            course_overviews = cls.get_select_courses(course_keys)
+            CourseOverviewGeneratedHistory.objects.create(num_courses=len(course_overviews))
```

Comment:
@robrap 
*> Is it only manually (or when empty) that this cache gets updated? Is there a planned scheduled update or some other determination that refresh is required?*

The cache is lazily either 
(1) populated one by one with each course when `get_from_id` is called, 
(2) populated in bulk when `get_select_courses` is called, or 
(3) seeded completely when `get_all_courses` is called.  

The course catalog page and API require the last of the 3.  They need to make sure that the cache was completely seeded before they can rely on the cache's data as the source of truth.  That is why I introduced the `CourseOverviewGeneratedHistory`.  So I can tell whether #3 happened.

The cache can also be populated/seeded preemptively by calling the `generate_course_overview` management command.

*> Are there any concerns regarding the the performance of this refresh? *

Yes.  On sites with a large number of courses.  See my response above regarding edx.org and open edX instances.